plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -82,6 +85,41 @@ def sync_feed(self):
             repo_url += '/'
         return repo_url
 
+    @contextlib.contextmanager
+    def update_state(self, state_dict, unit_type=None):
```

Comment:
This cleanup of the way the ``run()`` method manages the state of steps enabled me to add the additional skipping logic without that method getting unreasonably complex.